tests/test.py

```diff
@@ -83,51 +82,82 @@ def setUp(self):
         self.ec2_mock.return_value = self.ec2_connect_result
         boto.ec2.connect_to_region = self.ec2_mock
 
-        self.cfn_mock = mock.Mock(name="boto.cloudformation.connect_to_region")
-        self.cfn_connect_result = mock.Mock(name='cf_connect')
-        self.cfn_mock.return_value = self.cfn_connect_result
-        boto.cloudformation.connect_to_region = self.cfn_mock
+        cfn_mock = mock.Mock(name="boto.cloudformation.connect_to_region")
+        self.cfn_conn_mock = mock.Mock(name='cf_connect')
+        cfn_mock.return_value = self.cfn_conn_mock
+        boto.cloudformation.connect_to_region = cfn_mock
 
-    def test_get_stack_instance_ids(self):
-        scaling_group = mock.Mock()
-        rt = mock.PropertyMock(
-            return_value='AWS::AutoScaling::AutoScalingGroup')
-        sgi = mock.PropertyMock(
-            return_value='some-resource-id')
-        type(scaling_group).resource_type = rt
-        type(scaling_group).physical_resource_id = sgi
-
-        stack_mock = mock.Mock()
-        mock_config = {'list_resources.return_value': [scaling_group]}
-        stack_mock.configure_mock(**mock_config)
-
-        cf_mock = mock.Mock()
-        cf_connect_result = mock.Mock(name='cf_connect')
-        cf_mock.return_value = cf_connect_result
-        mock_config = {'describe_stacks.return_value': [stack_mock]}
-        cf_connect_result.configure_mock(**mock_config)
-        boto.cloudformation.connect_to_region = cf_mock
+    def test_get_stack_id(self):
+        stack_id = "arn:aws:cloudformation:eu-west-1:123/stack-name/uuid"
 
-        instance_mock = mock.Mock()
-        instance_id = mock.PropertyMock(return_value='i-12345')
-        type(instance_mock).instance_id = instance_id
+        stack = mock.Mock()
+        type(stack).stack_id = stack_id
+        self.cfn_conn_mock.describe_stacks.return_value = [stack]
+
+        cfn = cloudformation.Cloudformation(self.env.aws_profile, 'aws_region')
+        self.assertEqual(cfn.get_stack_id("stack-name"), stack_id)
+        self.assertEqual(cfn.get_stack_id(stack_id), stack_id)
+
+    def test_filter_stack_instances_when_no_stack(self):
+        stack_name = "our-stack"
+        stack_id = "aws:arn:stack-id"
+        cfn = cloudformation.Cloudformation(self.env.aws_profile, 'aws_region')
+
+        cfn.get_stack_id = mock.Mock(return_value=stack_id)
+
+        cfn.conn_ec2.get_all_reservations = mock.Mock(return_value=[])
+
+        self.assertEqual(cfn.filter_stack_instances(stack_name, {}), [])
+
+        self.assertEqual(
+            cfn.conn_ec2.get_all_reservations.get_all_reservations.called,
+            False,
+            "get_all_reservations should not be called")
+
+    def test_filter_stack_instances(self):
```

Comment:
Good call - I've added a doc string to test to make it clearer what we're doing, and added an mock reservation with one instance that we test